### ISSUE TITLE
[Suggestion] Speed of rails is always the same regardless of scale

### ISSUE THREAD

**Issue by CoasterBlock** [tsunamayo/Starship-EVO#5200]
Regardless if a rail-brick is at 1/8th m or 32m it always travels at the same rate, instead of proportional to the grids scale. 
This also applies to pistons. 

Would it be possible to make rail-speed adjustable like with pistons/rotors/hinges? And could the speed be proportional the scale of the used brick? 

Example: a 16m long rail at 1m scale would travel at 1m/s. 
A rail built using 16m bricks would travel at 16m/s. This way a rail that is built using anything larger than a 1m brick scaling would not take actual hours to move its entire distance. 
Alternatively, have the speed be fully adjustable in  the GUI. Instead of a slider, a value could be typed in. Anywhere from 1m/s to lets say 32m/s

**Comment by piotmetans** [comment]
I want to chime in on this topic because I've found a _really cool_ way to use rails in their current form that a change like this would completely break. 

Currently, rails actually move when you activate them, which can be read by an event gate. This allows for some really awesome touch-screen controls like this:


https://github.com/tsunamayo/Starship-EVO/assets/660015/c8fb54bc-8009-47ce-a707-1ed29d9e1d13


If you were to instead cause a menu to open up when activating a rail, like with the other mechanisms, this type of really cool in-depth building would not be possible. I understand that this is not the intended way to use rails, and that all mechanisms should probably work similarly from a UI standpoint, so I'd like to add another suggestion to coincide with CoasterBlock's: **Add some sort of activator block or blank screen block that functions similarly to the rail or a button from a  logic standpoint**. This keeps the current rail-screen tech while also maintaining UI cohesion across all mechanisms. Alternatively, a way to activate mechanisms player-side without opening a menu would work as well.

I also would like to mention that I understand that buttons already exist in the game, but the reality is that there are a lot of build styles that don't mesh very well with the large and cartoony buttons that we currently have (like in the example shown above), and the buttons aren't nearly as customizable as the current rail-screen tech that I'm showcasing here. Having an alternative to buttons (like activating blank screens) would be a godsend for logic-based mechanism nerds like myself!

**Comment by CoasterBlock** [comment]
I agree, I would like to keep that "button" funtion as well. It is convenient. Perhaps a work-around would simply involve having better speed control on rails. Either way, I think changes like these would need to be discussed with the community in general. Especially due to the examples you mentioned.

**Comment by Majikmonster** [comment]
I like the screen-button idea, but as far as mechanism movement speed, you'll find this suggestion much more befitting:
 #5012
